staging/src/k8s.io/cloud-provider/options/options.go

```diff
@@ -194,12 +194,10 @@ func (o *CloudControllerManagerOptions) ApplyTo(c *config.Config, userAgent stri
 		ClientConfig: c.Kubeconfig,
 	}
 	if c.ComponentConfig.KubeCloudShared.UseServiceAccountCredentials {
-		c.ClientBuilder = clientbuilder.SAControllerClientBuilder{
-			ClientConfig:         restclient.AnonymousClientConfig(c.Kubeconfig),
-			CoreClient:           c.Client.CoreV1(),
-			AuthenticationClient: c.Client.AuthenticationV1(),
-			Namespace:            metav1.NamespaceSystem,
-		}
+		c.ClientBuilder = clientbuilder.NewDynamicClientBuilder(
```

Comment:
This means the user running the cloud controller manager needs serviceaccount/token create permissions. This is included in the `system:kube-controller-manager` role... is that what the cloud controller manager process is expected to be bound to?

cc @jpbetz

Reply:
I believe the KCM has these permission because its running the ServiceAccountController. The CCM is not running the ServiceAccountController, so ideally it should not have or need these elevated permissions. Ideally I think we should find a way to separate the controllers which need these elevated permissions from the KCM to minimize the surface area of privilege escalation attack.

Reply:
SAController is not the reason that KCM needs serviceaccount/token create permission. some volumes related controller requires that permission, but more direct, https://github.com/kubernetes/kubernetes/blob/186f934e4ccb23e7dca03cc836f11f89e7d99e08/staging/src/k8s.io/cloud-provider/config/types.go#L55 requires that as well, which is shared by both controllers.

Reply:
> The CCM is not running the ServiceAccountController, so ideally it should not have or need these elevated permissions

It makes sense for CCM to be able to request tokens specifically for the service accounts for the control loops in the CCM. That is possible to grant in a narrow way (grant "serviceaccounts/tokens" on just the resourceNames for the expected service accounts), but I couldn't find where the CCM role is defined.

Reply:
I was (understandably) unable to convince anyone to put the CCM SA as part of the standard bootstrap as its not a core component of K8s. So its up to each cloud provider to set this up. So for example the GCP setup can be found at https://github.com/kubernetes/cloud-provider-gcp/blob/0e9c8d0151f46b5a6c0beea7778c034c6984d5d0/cluster/addons/cloud-controller-manager/cloud-node-controller-binding.yaml#L13.